cmd/helmhook/hook/multiversion/configconstraint.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multiversion
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	appsv1beta1 "github.com/apecloud/kubeblocks/apis/apps/v1beta1"
+	"github.com/apecloud/kubeblocks/cmd/helmhook/hook"
+	"github.com/apecloud/kubeblocks/pkg/client/clientset/versioned"
+	"github.com/apecloud/kubeblocks/pkg/constant"
+)
+
+const ccResource = "configconstraints"
+
+var configConstraintBeta1GVR = appsv1beta1.SchemeGroupVersion.WithResource(ccResource)
+
+func init() {
+	hook.RegisterCRDConversion(hook.NewNoVersion(0, 8), hook.NewNoVersion(0, 9), configConstraintBeta1GVR, &ccConversion{})
```

Comment:
User may upgrade from 0.7.x to 0.9 directly?

Reply:
done